crates/bevy_ecs/src/schedule/mod.rs

```diff
@@ -206,6 +206,14 @@ impl Schedule {
         // of the game. Closures inherit generic parameters from their enclosing function.
         #[cold]
         fn stage_not_found(stage_label: &dyn Debug) -> ! {
+            if ["PreStartup", "Startup", "PostStartup"]
+                .contains(&format!("{:?}", stage_label).as_str())
+            {
+                panic!(
+                    "Stage '{:?}' does not exist or is not a SystemStage.\nIt looks like you may be trying to add a startup system. Use `add_startup_system_to_stage()` instead of `add_system_to_stage()`.",
```

Comment:
```suggestion
                    "It looks like you may be trying to add a startup system. Use `add_startup_system_to_stage()` instead of `add_system_to_stage()`.",
```

The stage does exist, and is a `SystemStage`, so we should remove that from the error message entirely.

Reply:
My thinking was that the debug string of an enum is just the variant name, so it is _possible_ that the user has their own stage label variant called PreStartup, Startup, or PostStartup. If we're willing to consider that a negligible chance then your suggestion sounds good.

Reply:
If they're trying to use a custom stage label variant named `Startup` and it's not a startup stage, I think I'm okay to classify that one as "user error" and let them puzzle out ;) Perhaps we could expand the error message to include "Use `add_startup_system_to_stage(`StartupStage::'{:?}', my_system)`" to help guide them further?

I think we need better tools to configure stages in a more type-ful way to make that sort of issue impossible, but again, way out of scope for this fix.